### DISCUSSION TITLE
Inquiry about trainable pilot positions in Sionna

### DISCUSSION THREAD

**Discussion by xiaobaijh** [NVlabs/sionna#702] [General]
Hello,

Thank you for your excellent work on the Sionna project! I have a question regarding the implementation of trainable pilots.

From my understanding, Sionna currently supports trainable pilots by setting the pilot values as `tf.Variable`, which allows the values themselves to be optimized during training. However, in many related works, trainable pilots often refer to the ability to optimize pilot positions (i.e., the location of pilots in the resource grid), rather than just the pilot values.

I am wondering:

How to use the trainable pilot pattern ? Could it bring  communication performance gains compared to optimizing pilot positions?

Does Sionna currently support trainable pilot positions, or is this feature planned for future updates?
If not, could you suggest an approach for implementing trainable pilot positions in Sionna?

Looking forward to your guidance on this!

Thank you for your time and support.

Best regards

**Comment by SebastianCa** [comment]
Hi @xiaobaijh,
There is currently no dedicated API to optimize the pilot positions. However, you could write a custom resource grid and optimize the pilot positions.

An example for custom constellations can be found in the [end-to-end learning notebook](https://nvlabs.github.io/sionna/examples/Autoencoder.html), and a more advanced setup is shown in the [pilotless communications example](https://github.com/NVlabs/neural_rx/blob/main/notebooks/e2e_pilotless_communications.ipynb).

I’ll transfer this thread to discussions as it is not an issue with Sionna.
